src/parser/entities/mesh/parser.ts

```diff
@@ -0,0 +1,132 @@
+import { isMatched } from '@src/parser/shared';
+import type DxfArrayScanner from '../../DxfArrayScanner';
+import type { ScannerGroup } from '../../DxfArrayScanner';
+import {
+    createParser,
+    DXFParserSnippet,
+    Identity,
+    PointParser,
+    ToBoolean,
+    Trim,
+} from '../../shared/parserGenerator';
+import { CommonEntitySnippets } from '../shared';
+import type { MeshEntity } from './types';
+
+
+const MeshEntityParserSnippets: DXFParserSnippet[] = [
+    {
+        code: 90,
+        name: 'overridenSubEntityCount',
+        parser: Identity,
+    },
+    {
+        code: 140,
+        name: 'edgeCreaseWeights',
+        parser: Identity,
+        isMultiple: true,
+    },
+    {
+        code: 95,
+        name: 'edgeCreaseCount',
+        parser: Identity,
+    },
+    {
+        code: 94,
+        parser(curr, scanner, entity) {
+            // it assumes 90 followed by 94 directly
+            entity.edgeCount = curr.value
+            entity.edgeIndices = []
+
+            for (let i = 0; i < entity.edgeCount; ++i) {
+                const edgeIndices: number[] = [];
+
+                curr = scanner.next()
+                edgeIndices[0] = curr.value
+                curr = scanner.next()
+                edgeIndices[1] = curr.value
+
+                entity.edgeIndices.push(edgeIndices)
+            }
+        }
+    },
+    {
+        code: 93,
+        parser(curr, scanner, entity) {
+            // it assumes 90 followed by 93 directly
+            entity.totalFaceIndices = curr.value
+            entity.faceIndices = []
+
+            let count = 0
+
+            // read flattened indices safely then split
+            const rawIndices = [] as number[]
+
+            for (let i = 0; i < entity.totalFaceIndices && !isMatched(curr, 0); ++i) {
```

Comment:
Using `while` loop during parsing is very dangerous. As I can know how much entries are there, I used `for` loop for data fetching.